### PR TITLE
remove deprecated API for v0.8

### LINKED ISSUE #780: remove deprecated APIs [TBD for v0.8.0]
## 🚀 Feature

In version 0.5.3 and 0.6 was some API changed but for compatibility reasons we still keep it... 
This deprecated API shall be cleaned in upcoming 0.8 release :rocket:

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?
Fixes #780

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -52,6 +52,12 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Remove obsolete `self._device` in Trainer ([#1849](https://github.com/PyTorchLightning/pytorch-lightning/pull/1849))
```
Comment (Borda) [inline CHANGELOG.md:53]:
```suggestion
- Removed obsolete `self._device` in Trainer ([#1849](https://github.com/PyTorchLightning/pytorch-lightning/pull/1849))
```

File: pytorch_lightning/trainer/auto_mix_precision.py [hunk 2]
```diff
@@ -20,12 +20,6 @@ class TrainerAMPMixin(ABC):
     use_native_amp: bool
 
     def init_amp(self, use_amp):
```
Comment (williamFalcon) [inline pytorch_lightning/trainer/auto_mix_precision.py:22]: ummm this needs to stay.
We can't drop this until pytorch 1.6 comes out

**Comment by Borda** [comment]
@PyTorchLightning/core-contributors mind review this, it is mandatory for v0.8.0 :rabbit:

File: pytorch_lightning/trainer/deprecated_api.py [hunk 3]
```diff
@@ -141,3 +37,17 @@ def num_tpu_cores(self):
         """Back compatibility, will be removed in v0.9.0"""
         rank_zero_warn("Argument `num_tpu_cores` is now set by `tpu_cores` since v0.7.6"
                        " and this argument will be removed in v0.9.0", DeprecationWarning)
+
+    @property
+    def amp_level(self) -> str:
```
Comment (williamFalcon) [inline pytorch_lightning/trainer/deprecated_api.py:42]: amp cannot be removed until PT 1.6

File: pytorch_lightning/trainer/trainer.py [hunk 4]
```diff
@@ -268,13 +250,11 @@ def __init__(
                     stored in a different place than the logs written in `default_root_dir`.
 
             amp_level: The optimization level to use (O1, O2, etc...).
+                .. warning:: .. deprecated:: 0.7.6
 
-            num_sanity_val_steps: Sanity check runs n batches of val before starting the training routine.
+                    Has no effect for native AMP. Will remove 0.9.0.
```
Comment (williamFalcon) [inline pytorch_lightning/trainer/trainer.py:255]: cannot remove amp

**Pull request merged**